src/parse.rs

```diff
@@ -125,31 +125,31 @@ impl<'a> Default for ItemBody {
 pub struct BrokenLink<'a> {
     pub span: std::ops::Range<usize>,
     pub link_type: LinkType,
-    pub reference: &'a str,
+    pub reference: CowStr<'a>,
```

Comment:
Is this change required to solve the issue? This seems like the greatest breaking change.

Reply:
Yes, this is required. Otherwise you run into lifetime issues in pulldown itself - see 'Unfortunately, this uncovers that the change can't work:' in the PR description.

There's another (hidden) issue here: if you don't specify the lifetimes to the closure, this will given an inference error:

```rust
        let mut closure = |link: BrokenLink| Some(("#".into(), link.reference.into()));
```
```
error: lifetime may not live long enough
    --> src/parse.rs:1775:46
     |
1775 |         let mut closure = |link: BrokenLink| Some(("#".into(), link.reference.into()));
     |                            ----            - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
     |                            |               |
     |                            |               return type of closure is Option<(strings::CowStr<'_>, strings::CowStr<'2>)>
     |                            has type `parse::BrokenLink<'1>`
```
but I don't think this is actually an issue because you couldn't return a reference to the input before anyway, that was the whole issue. So I don't think this is actually regression in practice, although maybe it should be documented.

If you don't return a reference it works fine/.